### PR TITLE
Allow to skip loading unused plugins in Docker container

### PR DESCRIPTION
**Pull request by nineinchnick**
## Description

> **Note** This is only a suggestion. I'm fine with closing this PR if this is not the best idea for any reason.

Sometimes users would like to completely avoid loading unused plugins. The best recommendation would be to rebuild the Docker image, but some might struggle with this. This PR introduces two new environmental variables recognized only by the Docker container: `TRINO_ENABLE_PLUGINS` and `TRINO_DISABLE_PLUGINS`, which are inclusive and exclusive comma-separated lists of plugin names. Only one of them can be non-empty.

Removing plugins doesn't affect the startup time much (there's about 1s of variance):
```
Baseline: 8.345482000
With tpch,tpcds only: 7.005982000
Without tpch,tpcds: 9.227644000
```

For curiosity, I checked for `arm64` (with emulation):
```
Baseline: 115.821223000
With tpch,tpcds only: 96.419130000
Without tpch,tpcds: 116.562944000
```

If adding environmental variables like this is not a good idea, maybe it's better to add some support at build time, by defining build args that would do the same. That would also be a chance to document how to build custom images based on the official Trino images.

> Is this change a fix, improvement, new feature, refactoring, or other?

> Is this a change to the core query engine, a connector, client library, or the SPI interfaces? (be specific)

> How would you describe this change to a non-technical end user or system administrator?

## Documentation

( ) No documentation is needed.
( ) Sufficient documentation is included in this PR.
( ) Documentation PR is available with #prnumber.
( ) Documentation issue #issuenumber is filed, and can be handled later.

## Release notes

( ) No release notes entries required.
( ) Release notes entries required with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

File: core/docker/bin/run-trino [hunk 1]
```diff
@@ -2,14 +2,9 @@
...
 set -xeuo pipefail
 
-set +e
-grep -s -q 'node.id' /etc/trino/node.properties
-NODE_ID_EXISTS=$?
-set -e
-
 NODE_ID=""
-if [[ ${NODE_ID_EXISTS} != 0 ]] ; then
+if ! grep -s -q 'node.id' /etc/trino/node.properties; then
```
Comment (findepi) [inline core/docker/bin/run-trino:6]: separate pr and let's have this merged

File: core/docker/bin/run-trino [hunk 2]
```diff
@@ -7,4 +7,34 @@ if ! grep -s -q 'node.id' /etc/trino/node.properties; then
...
+        echo >&2 "Cannot set both \$TRINO_DISABLE_PLUGINS and \$TRINO_ENABLE_PLUGINS"
+        exit 1
+    fi
+    mv "$base/plugin" "$base/plugin.disabled"
+    mv "/etc/trino/catalog" "/etc/trino/catalog.disabled"
+    mkdir -p "$base/plugin" /etc/trino/catalog
+    IFS=, read -ra names <<< "$TRINO_ENABLE_PLUGINS"
+    for name in "${names[@]}"; do
+        mv "$base/plugin.disabled/$name" "$base/plugin/$name"
+        if [ -f "/etc/trino/catalog.disabled/$name.properties" ]; then
```
Comment (findepi) [inline core/docker/bin/run-trino:25]: plugin name != catalog name

did you mean TRINO_ENABLED_CATALOGS env var?

File: core/docker/container-test.sh [hunk 3]
```diff
@@ -56,6 +63,8 @@ function test_container {
     local PLATFORM=$2
     echo "🐢 Validating ${CONTAINER_NAME} on platform ${PLATFORM}..."
     test_javahome "${CONTAINER_NAME}" "${PLATFORM}"
-    test_trino_starts "${CONTAINER_NAME}" "${PLATFORM}"
+    test_trino_starts -e TRINO_ENABLE_PLUGINS=jmx,memory --platform "${PLATFORM}" "${CONTAINER_NAME}"
+    test_trino_starts -e TRINO_DISABLE_PLUGINS=tpch,tpcds --platform "${PLATFORM}" "${CONTAINER_NAME}"
+    test_trino_fails -e TRINO_ENABLE_PLUGINS=jmx,memory -e TRINO_DISABLE_PLUGINS=tpch,tpcds "${CONTAINER_NAME}"
```
Comment (findepi) [inline core/docker/container-test.sh:68]: test_trino_fails should assert on the  "exception message"

File: core/docker/README.md [hunk 4]
```diff
@@ -19,7 +19,9 @@ To launch it, execute the following:
 docker run -p 8080:8080 --name trino trinodb/trino
 ```
 
-Wait for the following message log line:
+When Trino finishes starting up, the container will report its status as healthy.
+You can check it in the output of `docker ps`, or wait for the following message
+log line:
```
Comment (findepi) [inline core/docker/README.md:24]: unrelated

File: core/docker/bin/run-trino (hunk 2 above)
Comment (nineinchnick) [reply core/docker/bin/run-trino:25]: Whoops, you're right. I added this because there are default catalogs that will cause the server to fail at startup if the connectors/plugins they use are disabled. I want to keep this simple so I'd like to avoid adding even more env vars. Would grepping for `connector.name=$name` be ok instead?

File: core/docker/bin/run-trino (hunk 1 above)
Comment (nineinchnick) [reply core/docker/bin/run-trino:6]: #12487

File: core/docker/bin/run-trino (hunk 2 above)
Comment (nineinchnick) [reply core/docker/bin/run-trino:25]: That's another reason to only do this at build time - to avoid adding new logic to parse catalog configs outside of Java.

**Comment by hashhar** [comment]
I don't have strong opinions about this but IMO I don't see a lot of benefits here - the startup time isn't affected much. The only thing it might help with is satisfying security scanners which we should do the proper way.

This does allow people to break things if they don't know what they are doing by for example excluding the geospatial plugin which is a dependency of Hive connector, or shared plugins like resource-group-managers/password-authenticators/teradata-functions.
